### ISSUE TITLE
Private replies from non-OP users emailed to OP

### ISSUE THREAD

**Issue by strangerstudios** [pippinsplugins/bbPress-Private-replies#25]
We've had threads where other non-moderator/non-OP users posted private replies. These private replies are hidden from the thread author (OP) when viewing the forums (good IMO) but are still emailed to the OP if the OP subscribes to the thread.

I believe the email behavior should match the behavior of what is hidden in the forum.

To do this, the second half of this conditional be removed:
https://github.com/pippinsplugins/bbPress-Private-replies/blob/master/bbp-private-replies.php#L274

So just `if( user_can( $user_id, 'moderate' )`

**Comment by asadkn** [comment]
This can be a huge security/privacy issue if the a second user shares some private data - assuming it will be only shared with moderators and not with the original poster. It definitely needs to be fixed.

**Comment by pippinsplugins** [comment]
This fix is incorrect. That will prevent the OP from receiving emails when a moderator posts a reply.

With the current logic, I don't see how a non-OP or non-moderator would ever receive that email.

**Comment by asadkn** [comment]
I realized that the fix is wrong but forgot to update it. Basically it was for this scenario:

- Joe starts a topic and a moderator asks Joe to share private info.
- Shane comes along in the same topic and shares her private info too. The OP (Joe) is also sent this info.

The solution to that problem is to only notify the OP when a moderator sends a private reply. I realized later it may not be ideal for every case so didn't post it again. But it's definitely an issue for Support Forums where the private info is only to be shared with moderators.

I now use something along the lines of:

```
			// Make sure the private reply is by a moderator before notifying the OP
			$should_notify_op = user_can( $reply_author, 'moderate' ) && (int) $topic_author === (int) $user_id;
			
			if( user_can( $user_id, 'moderate' ) || $should_notify_op ) {
```

Perhaps the real solution would be to add it as an option or add a filter for it.
